pandas-stubs/core/indexes/accessors.pyi

```diff
@@ -434,49 +442,43 @@ class TimedeltaIndexProperties(
     _DatetimeRoundingMethods[TimedeltaIndex],
 ): ...
 
-class _dtDescriptor(CombinedDatetimelikeProperties, Generic[S1]):
+@type_check_only
+class DtDescriptor:
     @overload
-    def __get__(self, instance: Series[Never], owner: Any) -> Never: ...
+    def __get__(self, instance: Series[Never], owner: type[Series]) -> Never: ...
```

Comment:
I think we should remove this.  If someone has an untyped `Series`, and they want to use a `dt` method, then they should cast the untyped `Series` to `Series[Timestamp]` or `Series[Timedelta]` to tell the type checker that they know what they are doing.

I know this is different than what we did with the arithmetic, but I think `.dt` (and `.str` for that matter) are different because they will fail if you apply them to a Series with the wrong dtype.

Reply:
Yes, so we should figure out how to deal with that.  Maybe return `Never` or an empty properties class.